ERR: qcut uniquess checking

ashishsingal1 opened the pull request:
- [x] closes #7751
- [ ] tests added / passed
- [ ] passes `git diff upstream/master | flake8 --diff`
- [ ] whatsnew entry

Add option to drop non-unique bins.

jankatins committed:
COMPAT: Require a problem description in issues

Currently some issues are just code examples without a description
**why** a change should be done. This leads to problems when
(years later) the current behaviour is questioned and no one can
remember why it was changed.

closes #14032

Commit:
Move compression code to io.common._get_handle

xref #14576
closes #13340

paul-mannino committed:
BUG: Fix issue with inserting duplicate columns in a dataframe

closes #14291
closes #14431

Commit:
BUG: to_numeric downcast = 'unsigned' would not un-sign a 0 value

closes #14504
closes #14401

ashishsingal1 committed:
Update tile.py

ashishsingal1 committed:
Update tile.py

sinhrks commented:
Thx for the PR. Can u add tests and whatsnew?

dubourg committed:
Type codes and categories as lists instead of tuples in _factorize_from_iterables (fixes #14438) (#14449)

ivallesp committed:
ERR: Checks for left_index and right_index merge parameters

Author: Iván Vallés Pérez <ivanvallesperez@gmail.com>

Closes #14434 from ivallesp/add-check-for-merge-indices and squashes the following commits:

e18b7c9 [Iván Vallés Pérez] Add some checks for assuring that the left_index and right_index parameters have correct types. Tests added.

jreback commented on pandas/tools/tile.py:
```diff
@@ -141,6 +142,9 @@ def qcut(x, q, labels=None, retbins=False, precision=3):
         as a scalar.
     precision : int
         The precision at which to store and display the bins labels
+    duplicate_edges : {'raise', 'drop'}, optional
```
duplicate_edges ->  ``errors='raise', 'drop'``

jreback replied:
actually, I like ``duplicates=raise, drop``

jreback commented on pandas/tools/tile.py:
```diff
@@ -191,7 +197,11 @@ def _bins_to_cuts(x, bins, right=True, labels=None, retbins=False,
     ids = bins.searchsorted(x, side=side)
 
```
check all the valid possibilities for ``errors`` and raise otherwise (IOW, if you pass a bad value should raise an informative message)

```
if errors not in ['raise', 'drop']:
   raise ValueError("invalid value for errors paramters, valid are: raise, drop")
```

jreback commented on pandas/tools/tile.py:
```diff
@@ -191,7 +197,11 @@ def _bins_to_cuts(x, bins, right=True, labels=None, retbins=False,
-        raise ValueError('Bin edges must be unique: %s' % repr(bins))
+        if (duplicate_edges == 'raise'):
+            raise ValueError('Bin edges must be unique: %s'
+                             % repr(bins))
```
expand this message to say, you can force edges to be unique by passing ``errors='drop'``

jreback commented:
looks pretty good. some more error checking formatting is needed. pls add a whatsnew note in 0.19.1, make an Enhancements changes section. Be sure to say that the default is the existing behavior.

jreback commented:
pls add some tests. use the example from the orginal issue exercising both options to ``duplicates`` and pass some invalid values as well (and assert those raise)

sinhrks committed:
BUG: pivot_table may raise TypeError without values

``pivot_table`` raises TypeError`` when ``index`` or ``columns`` is array-like and
``values`` is not specified.

Author: sinhrks <sinhrks@gmail.com>

Closes #14380 from sinhrks/pivot_table_bug and squashes the following commits:

be426db [sinhrks] BUG: pivot_table may raise TypeError without values

chris-b1 committed:
BUG: underflow on Timestamp creation (#14433)

* BUG: underflow on Timestamp creation

* undo change to lower bound

* change lower bound; but keep rounding to us

ashishsingal1 commented:
Thanks for the feedback -- this is my first PR on an open source project -- will add tests and resubmit tomorrow. Had some trouble building my branch on Windows.

aixtools committed:
Update unpack_template.h (#14441)

USE_CASE_RANGE is a GNU C feature. This change will activate
USE_CASE_RANGE on any platform when using GNU C and not on any platform
when a different compiler is being used.

closes #14373

jreback commented:
contributing docs are [here](http://pandas.pydata.org/pandas-docs/stable/contributing.html); there is a section on creating a windows env.

chris-b1 committed:
DOC: update readme for repo move (#14470)

gfyoung committed:
BUG: Catch overflow in both directions for checked add (#14453)

1) Add checks to ensure that add overflow does not
   occur both in the positive or negative directions.
2) Add benchmarks to ensure that operations involving
   this checked add function are significantly impacted.

jorisvandenbossche committed:
DOC: correct DataFrame.pivot docstring (#14430)

The mention of panels that are created is not correct. You get a multi-index

sinhrks committed:
BUG: String indexing against object dtype may raise AttributeError (#14424)

chris-b1 committed:
BLD: don't require cython on sdist install

closes #14475
closes #14204

gfyoung committed:
MAINT: Use check_output when merging.

Since we don't support Python 2.6 anymore, the `check_output` method
from `subprocess` is  at our disposal.    Follow-up to #14447.    xref
<a href="https://github.com/pandas-
dev/pandas/issues/14439#issuecomment-254522055"> #14439 (comment)</a>

Author: gfyoung <gfyoung17@gmail.com>

Closes #14465 from gfyoung/merge-pr-refactor and squashes the following commits:

e267d2b [gfyoung] MAINT: Use check_output when merging.

tserafim committed:
ERR: Fix GH13139: better error message on invalid pd.eval and df.query input

closes #13139

Added test case to check for invalid input(empy string) on pd.eval('')
and df.query('').  Used existing helper function(_check_expression)

Author: Thiago Serafim <thiago.serafim@gmail.com>

Closes #14473 from tserafim/issue#13139 and squashes the following commits:

77483dd [Thiago Serafim] ERR: correctly raise ValueError on empty input to pd.eval() and df.query() (#13139)
9a5c55f [Thiago Serafim] Fix GH13139: better error message on invalid pd.eval and df.query input

jorisvandenbossche committed:
BUG: fix empty intersection of RangeIndex (GH14364)

closes #14364

Author: Joris Van den Bossche <jorisvandenbossche@gmail.com>

Closes #14481 from jorisvandenbossche/bug-rangeindex-empty and squashes the following commits:

823e83d [Joris Van den Bossche] BUG: fix empty intersection of RangeIndex (GH14364)

keshavramaswamy committed:
DOC: updated docstring in .to_datetime() for out-of-bounds timestamps
with errors='ignore'

closes #14448

Author: Keshav Ramaswamy <keshav.ramaswamy@gmail.com>
Author: Keshav Ramaswamy <keshav.ramaswamy@infor.com>

Closes #14452 from keshavramaswamy/master and squashes the following commits:

5468fc5 [Keshav Ramaswamy] added link to timestamp limitations in docstring
3aa78cf [Keshav Ramaswamy] edit docstring to fit python standards
8bfa58e [Keshav Ramaswamy] edited docstring to fit python standards
5ed8ef5 [Keshav Ramaswamy] edut docstring to fit Python Standards
7402de4 [Keshav Ramaswamy] edited docstring to fit python standards
c16ad6b [Keshav Ramaswamy] added timestamp limitations and default behavior to docstring
0db07b5 [Keshav Ramaswamy] updated docstring in .to_datetime()

Liam3851 committed:
BUG: GH14323 Union of differences from DatetimeIndex incorrect

closes #14323

Sets freq to None when doing a difference operation on a DatetimeIndex
or TimedeltaIndex, rather than retaining the frequency (which can
cause  problems with downstream operations). Frequency of PeriodIndex
is retained.

Author: David Krych <davidk@ciphercap.com>

Closes #14346 from Liam3851/dtind_diff_14323 and squashes the following commits:

1dbf582 [David Krych] BUG: GH14323 Union of differences from DatetimeIndex incorrect

Commit:
BUG: block mutation of read-only array in series

Author: Joe Jevnik <joe@quantopian.com>

Closes #14359 from llllllllll/series-setitem and squashes the following commits:

9925327 [Joe Jevnik] BUG: fix a bug in Series.__setitem__ that allowed the mutatation of read-only arrays

Commit:
BUG: downcast='unsigned' on 0 would would not downcast to unsigned.

closes #14401

Author: Nicholas Ver Halen <verhalenn@yahoo.com>

Closes #14472 from verhalenn/issue14401 and squashes the following commits:

21e1a97 [Nicholas Ver Halen] Made the downcast limits test its own function.
a62a3f8 [Nicholas Ver Halen] Revert "Added release note to issue 14401 resolve."
7413a9c [Nicholas Ver Halen] Did some editing for some lint problems.
26fe3a1 [Nicholas Ver Halen] Made the dictionaries into one list of sets.
5023fc7 [Nicholas Ver Halen] Changed the test to work with python 3.x
9ccc991 [Nicholas Ver Halen] Changed the tests so that it iterated through a dictionary.
ef35e19 [Nicholas Ver Halen] Added tests for the max and min values of all dtypes to to_numeric
cc278ff [Nicholas Ver Halen] Added release note to issue 14401 resolve.
11a421d [Nicholas Ver Halen] Added a test to check uint8 with 0
11038f8 [Nicholas Ver Halen] Made it so that 0 was included in uint8

Commit:
Bug: Error when key-only Grouper is passed to groupby in a list (GH14334)

closes #14334

Author: Jon M. Mease <jon.mease@jhuapl.edu>

Closes #14342 from jmmease/bug_14334 and squashes the following commits:

5e96797 [Jon M. Mease] Add tests for grouping on two columns
cee5ce6 [Jon M. Mease] Added bug description to new test case
f9ef05b [Jon M. Mease] Moved whatsnew to 0.19.1 and clarified description
14a4ae6 [Jon M. Mease] Added whatsnew for GH 14334
9805c30 [Jon M. Mease] Fix for GH 14334
dfd3e09 [Jon M. Mease] Added test case for GH 14334

jreback committed:
BUG: incorrect broadcasting that could casuse dtype coercion in a groupby-transform

closes #14457

Author: Jeff Reback <jeff@reback.net>

Closes #14466 from jreback/transform and squashes the following commits:

ce595b9 [Jeff Reback] BUG: incorrect broadcasting that could casuse dtype coercion in a groupby-transform

jreback committed:
Revert "BUG: downcast='unsigned' on 0 would would not downcast to unsigned."

This reverts commit 6ff53c2b47b026f605e415d3cd5f3b0dda7e0774.

robertwb committed:
BLD: Support Cython 0.25

closes #14496

laudney committed:
PERF: performance regression in Series.asof (#14476)

* Fix performance regression in Series.asof by avoiding pre-computing nulls and returning value by indexing the underlying ndarray.

jreback committed:
COMPAT/TST: fix test for range testing of negative integers to neg powers

xref https://github.com/numpy/numpy/pull/8127    closes #14489

Author: Jeff Reback <jeff@reback.net>

Closes #14498 from jreback/compat and squashes the following commits:

882872e [Jeff Reback] COMPAT/TST: fix test for range testing of negative integers to neg powers

gfyoung committed:
BUG: Accept unicode quotechars again in pd.read_csv

Title is self-explanatory.  Affects Python 2.x only.  Closes #14477.

Author: gfyoung <gfyoung17@gmail.com>

Closes #14492 from gfyoung/quotechar-unicode-2.x and squashes the following commits:

ec9f59a [gfyoung] BUG: Accept unicode quotechars again in pd.read_csv

jreback committed:
BLD: fix 3.4 build for cython to 0.24.1

jorisvandenbossche committed:
TST: simplify tests for GH14346 (#14502)

cswarth committed:
DOC: Expand on reference docs for read_json() (#14442)

jorisvandenbossche committed:
BUG: fix DatetimeIndex._maybe_cast_slice_bound for empty index (GH14354) (#14501)

gfyoung committed:
MAINT: Expand lint for *.py (#14516)

jorisvandenbossche committed:
BUG/ERR: raise correct error when sql driver is not installed (#14527)

When the driver was not installed, but sqlalchemy itself was, when passing a URI string, you got an error indicating that SQLAlchemy was not installed, instead of the driver not being installed. This was because the import error for the driver was captured as import error for sqlalchemy.

parthea committed:
DOC: Simplify the gbq integration testing procedure for contributors (#14541)

tworec committed:
BUG: tseries ceil doc fix (#14543)

gfyoung committed:
BUG: Don't parse inline quotes in skipped lines (#14514)

Closes gh-14459.

brandonmburroughs committed:
BUG: Dataframe constructor when given dict with None value (#14392)

jorisvandenbossche committed:
Update ISSUE_TEMPLATE: be more explicit on where to paste the output of show_versions

jreback committed:
asv compat for py3

jorisvandenbossche committed:
BUG: don't close user-provided file handles in C parser (GH14418) (#14520)

jorisvandenbossche committed:
BUG: DataFrame.quantile with NaNs (GH14357) (#14536)

jorisvandenbossche committed:
PERF: casting loc to labels dtype before searchsorted (#14551)

jorisvandenbossche committed:
DEPR: add deprecation warning for com.array_equivalent (#14567)

pandas.core.common.array_equivalent was removed without deprecation warning.
This commits adds it back to the core.common namespace with deprecation warning

jorisvandenbossche committed:
DOC: rst fixes

jorisvandenbossche committed:
BUG/API: Index.append with mixed object/Categorical indices (#14545)

* BUG/API: Index.append with mixed object/Categorical indices

* Only coerce to object if the calling index is not categorical

* Add test for the df.info() case (GH14298)

jorisvandenbossche committed:
DOC: update whatsnew/release notes for 0.19.1 (#14573)

dhimmel committed:
DOC: Update GitHub org from pydata to pandas-dev (#14575)

Revert pydata/pandas-datareader references

keshavramaswamy committed:
DOC: add sep argument to read_clipboard signature (#14537)

mroeschke committed:
DOC: Ordering during set operation on index (#3123) (#14629)

bashtage committed:
ENH: Explicit range checking of floats when writing Stata

Add explicit error checking for out-of-range doubles when writing Stata files
Upcasts float32 to float64 if out-of-range values encountered
Tests for infinite values and raises if found

closes #14618
closes #14637

dhimmel committed:
TST: Create compressed salary testing data (#14587)

maxalbert committed:
DOC/ERR: improve readability of error message (#14597)

Commit:
ENH: return an OrderedDict from read_excel with sheetname=None (#14571)

jorisvandenbossche committed:
TST: correct url for test file on s3 (xref #14587)

jreback committed:
DOC: setup for 0.19.2

jreback committed:
BUG: segfault manifesting with dateutil=2.6 w.r.t. replace when timezones are present

closes #14621

Author: Jeff Reback <jeff@reback.net>

Closes #14631 from jreback/replace and squashes the following commits:

3f95042 [Jeff Reback] BUG: segfault manifesting with dateutil=2.6 w.r.t. replace when timezones are present

MykolaGolubyev committed:
TST: Fix trailing current date zeros flaky test_format problem

closes #14626

Author: Mykola Golubyev <Mykola.Golubyev@twosigma.com>

Closes #14638 from MykolaGolubyev/issue-14626 and squashes the following commits:

166ec23 [Mykola Golubyev] Inline datetime_now_without_trailing_zeros within test_format
68a72a7 [Mykola Golubyev] Fix trailing current date zeros flaky test_format problem

jreback committed:
TST: skip test_gbq.test_upload_data_if_table_exists_replace for now

TomAugspurger committed:
COMPAT: Cast to string before raise in read_stata (#14657)

Fix ValueError message in StataReader

Cast to string before joining the list of typecodes in the error message.

jreback committed:
BLD: fix linting check for .pyx

closes #14659

Commit:
DOC: Update Series.mode docstring (#14653)

The sort keyword has never existed, and output is always sorted

bllchmbrs committed:
fix ##14664 (#14665)

mroeschke committed:
DOC: Explain pivot vs. pivot_table (#6950) (#14650)

Move pivot paragraph to pivot table section

add pivot_table reference to pivot

llllllllll committed:
BUG: don't allow users to move from an interned string (#14494)

mroeschke committed:
BUG: Bug upon Series.Groupby.nunique with empty Series

closes #12553
closes #14770

luca-s committed:
BUG: pandas.cut and negative values #14652

closes #14652

Author: Luca Scarabello <luca.scarabello@sed.ethz.ch>
Author: Luca <luca.scarabello@gmail.com>

Closes #14663 from luca-s/issue_14652 and squashes the following commits:

8db26db [Luca Scarabello] Moved new test location to pandas\tools\tests\test_tile.py
90dd07d [Luca Scarabello] Updated whatsnew
d6b3da8 [Luca Scarabello] fixed flake8 compliance
fdc55b9 [Luca Scarabello] Added test case for #14652
d5790e2 [Luca Scarabello] updated whatsnew v0.19.2
2db0c7a [Luca] BUG: pandas.cut and negative values #14652

jreback committed:
DOC: add whitespace to bug fix section of 0.19.2

jreback committed:
BLD: add py3.6 build
generic support for additional installation instructions (e.g. 3.4_slow build)
remove 2.7_NUMPY_DEV build files (not longer on travis)
use python 3.5 for root conda env, use PYTHON_VERSION to designate conda installation

jreback committed:
ERR: fix exception propogation for datetime parsing functions, noted in python 3.6

closes #14561

Author: Jeff Reback <jeff@reback.net>

Closes #14678 from jreback/py3.6_fix and squashes the following commits:

c8eed83 [Jeff Reback] ERR: fix exception propogation for datetime parsing functions, noted in python 3.6

jreback committed:
BLD: use miniconda3 as the miniconda dir

jreback committed:
BLD: cleaner 3.6 deps install

jreback committed:
BLD: use correct path for travis

jreback committed:
COMPAT: remove some deprecation warnings in 3.6

partial on #14679

Author: Jeff Reback <jeff@reback.net>

Closes #14681 from jreback/warnings and squashes the following commits:

56e7b1b [Jeff Reback] COMPAT: remove some deprecation warnings in 3.6

jreback committed:
ERR: more informative message on invalid Timestamp input

TST: fix unordable error message

xref #14679

TST: handle unorderable exceptions in indexing

closes #14684

jreback committed:
BUG: fix pickling of Custom offsets in 3.6

xref #14679

Author: Jeff Reback <jeff@reback.net>

Closes #14685 from jreback/offsets and squashes the following commits:

8ad212c [Jeff Reback] BUG: fix pickling of Custom offsets in 3.6

jreback committed:
COMPAT: pickle compat for Timestamp in py3.6
BUG: fix unorderable exception types in py3.6

closes #14689

jreback committed:
DOC: compat notice for 3.6

gfyoung committed:
API: Rename CParserError to ParserError (#14479)

Partially resolves gh-12665. We will remove
CParserError in the future.

Commit:
BUG in clipboard (linux, python2) with unicode and separator (GH13747)

vendered updated version of Pyperclip

closes #13747
closes #14362
closes #12807
closes #12529

Author: Ajay Saxena <ajasaxen@Ajays-MacBook-Pro.local>
Author: Ajay Saxena <aileronajay@gmail.com>

Closes #14599 from aileronajay/master and squashes the following commits:

2aafb66 [Ajay Saxena] moved comment inside test and added github issue labels to test
b74fbc1 [Ajay Saxena] ignore lint test for pyperclip files
9db42d8 [Ajay Saxena] whatsnew conflict
1dca292 [Ajay Saxena] conflict resolution
98b61e8 [Ajay Saxena] merge conflict
cedb690 [Ajay Saxena] merge conflict in whats new file
7af95da [Ajay Saxena] merging lastest changes
ac8ae60 [Ajay Saxena] skip clipboard test if clipboard primitives are absent
b03ed56 [Ajay Saxena] changed whatsnew file
c0aafd7 [Ajay Saxena] Merge branch 'test_branch'
9946fb7 [Ajay Saxena] Merge branch 'master' of https://github.com/pandas-dev/pandas into test_branch
ed1375f [Ajay Saxena] Merge branch 'test_branch'
0665fd4 [Ajay Saxena] fixed linting and test case as per code review
d202fd0 [Ajay Saxena] added test for valid encoding, modified setup.py so that pandas/util/clipboard can be found
dd57ae3 [Ajay Saxena] code review changes and read clipboard invalid encoding test
71d58d0 [Ajay Saxena] testing encoding in kwargs to to_clipboard and test case for the same
02f87b0 [Ajay Saxena] removed duplicate files
825bbe2 [Ajay Saxena] all files related to pyperclip are under pandas.util.clipboard
c5a87d8 [Ajay Saxena] Merge branch 'test_branch' of https://github.com/aileronajay/pandas into test_branch
f708c2e [Ajay Saxena] Merge branch 'master' of https://github.com/aileronajay/pandas
d565b1f [Ajay Saxena] updated pyperclip to the latest version
14d94a0 [Ajay Saxena] changed the pandas util clipboard file to return unicode if the python version is 2, else str
66d8ebf [Ajay Saxena] removed the disabled tag for clipboard test so that we can check if they pass after this change
edb8553 [Ajay Saxena] refactored the new unicode test to be in sync with the rest of the file
c83d000 [Ajay Saxena] added test case for unicode round trip
fb922d6 [Ajay Saxena] changes for GH 13747

mroeschke committed:
BUG: _nsorted incorrect with duplicated values in index

closes #13412
closes #14707

Commit:
BUG: Fix move_into_mutable_buffer for python 3.6.

In python 3.6, the CALL_FUNCTION handling was updated. One change is
that when calling a C function from a python function python now
counts  the reference owned by the argument tuple. This means that
move was  always seeing objects with two references instead of the
expected  one.    Python 3.6 also removed a copy in the argument tuple
when *unpacking  functions. This means that if a user does:    tuple =
(create_string(),)  move_into_mutable_buffer(*tuple)    where
create_string() creates a string object with one reference then we
will fail to raise a BadMove even though the user could later retrieve
that string with tuple[0]. There is no way to detect this case so this
patch adds a warning to the docstring advising against star unpacking.
xref: #14679    I played around with removing the extra reference that
was added in 3.6 but it looks like playing with borrowed refs
everywhere will be a bit tricky. This change should clear things up
for 3.6 while continuing to work for older versions. In 3.6 you
__could__ get a shared mutable string from this but you need to try
pretty hard for it.

Author: Joe Jevnik <joe@quantopian.com>

Closes #14695 from llllllllll/move-3.6-compat and squashes the following commits:

d1e8b1b [Joe Jevnik] BUG: Fix move_into_mutable_buffer for python 3.6.

Commit:
BUG: Fix parse empty df

closes #14515

This commit fixes a bug where `read_csv` failed when given a file with
a multiindex header and empty content. Because pandas reads index
names as a separate line following the header lines, the reader looks
for the line with index names in it. If the content of the dataframe
is empty, the reader will choke. This bug surfaced after
https://github.com/pandas-dev/pandas/issues/6618 stopped writing an
extra line after multiindex columns, which led to a situation where
pandas could write CSV's that it couldn't then read.     This commit
changes that behavior by explicitly checking if the index name row
exists, and processing it correctly if it doesn't.

Author: Ben Kandel <ben.kandel@gmail.com>

Closes #14596 from bkandel/fix-parse-empty-df and squashes the following commits:

32e3b0a [Ben Kandel] lint
e6b1237 [Ben Kandel] lint
fedfff8 [Ben Kandel] fix multiindex column parsing
518982d [Ben Kandel] move to 0.19.2
fc23e5c [Ben Kandel] fix errant this_columns
3d9bbdd [Ben Kandel] whatsnew
68eadf3 [Ben Kandel] Modify test.
17e44dd [Ben Kandel] fix python parser too
72adaf2 [Ben Kandel] remove unnecessary test
bfe0423 [Ben Kandel] typo
2f64d57 [Ben Kandel] pep8
b8200e4 [Ben Kandel] BUG: read_csv with empty df

jsantander committed:
BUG: Avoid AmbiguousTimeError on groupby

closes #14682

Author: Julian Santander <julian.santander@nokia.com>
Author: Julian Santander <jsantander2@gmail.com>

Closes #14683 from j-santander/master and squashes the following commits:

d90afaf [Julian Santander] Addressing additional code inspection comments
817ed97 [Julian Santander] Addressing code inspections comments
99a5367 [Julian Santander] Fix unittest error and lint warning
940fb22 [Julian Santander] Avoid AmbiguousTimeError on groupby

mroeschke committed:
TST: Test aggregation over arrays (#3788)

closes #3788

Author: Matt Roeschke <emailformattr@gmail.com>

Closes #14675 from mroeschke/fix_3788 and squashes the following commits:

01dce9d [Matt Roeschke] TST: Test aggregation over arrays (#3788)

Commit:
DOC: update FAQ to note pandas-qt only works for python 2.x (#14713)

jreback committed:
TST: skip test_transactions in sqlitefallback on py3.6

mroeschke committed:
DOC: Disambiguate 'where' in boolean indexing-10min.rst (#12661) (#14708)

Expand on boolean indexing example

dwillmer committed:
DOC: fix typo in merge_asof docstring examples (#14718)

nicholsn committed:
doc: comverted --> converted (#14722)

gfyoung committed:
DEPR: Patch to_dense behaviour for sparse.

Patches the following for `to_dense`:    1) Fix `SparseArray.to_dense`
documentation to refer to `SparseArray` and not `SparseSeries`.    2)
Deprecate the fill parameter in `SparseArray.to_dense`, as that
parameter was not being respected.    3) Deprecate the sparse_only
parameter in `SparseSeries.to_dense`, as that parameter is
inconsistent with the `to_dense` API we want, which is no parameters.
Closes #14647.

Author: gfyoung <gfyoung17@gmail.com>

Closes #14686 from gfyoung/to-dense-patch and squashes the following commits:

ad7da32 [gfyoung] BUG: Patch to_dense behaviour for sparse.

sdimi committed:
Update frame.py (#14724)

typo "explicitly"

gfyoung committed:
BUG: Respect the dtype parameter for empty CSV (#14717)

Commit:
DOC: Correct uniqueness of index for Series  (#14344)

mattip committed:
CLN: move assignment from header into cython (#14731)

gfyoung committed:
MAINT: Ignore .pxi files

All `.pxi` files have the warning to not edit directly, so why not
have Git ignore any changes made there?    In addition, this PR
deletes all of the `.pxi` files from the codebase because they are
automatically generated by the `.pxi.in` files

Author: gfyoung <gfyoung17@gmail.com>

Closes #14723 from gfyoung/gitignore-pxi-ignore and squashes the following commits:

fe0c410 [gfyoung] MAINT: Ignore .pxi files

jreback committed:
DOC: missing ref in timeseries.rst (#14745)

gfyoung committed:
BUG: Improve error message for multi-char sep and quotes in Python engine (#14582)

If there is a field counts mismatch, check whether
a multi-char sep was used in conjunction with quotes.
Currently, that setup is not respected and can result
in improper line breaks.

Closes gh-13374.

mroeschke committed:
DOC: Remove outdated caveats for Anaconda and HTML parsing (#9032) (#14739)

chris-b1 committed:
API: add dtype= option to python parser (#14295)

jorisvandenbossche committed:
BUG: mixed freq timeseries plotting with shared axes (GH13341) (#14330)

gfyoung committed:
Standardize function signatures (#14645)

Standardize the following function signatures:

1) repeat(reps, *args, **kwargs)
2) searchsorted(value, side='left', sorter=None)

Closes gh-12662.

jorisvandenbossche committed:
Revert "TST/TEMP: fix pyqt to 4.x for plotting tests" (#14744)

kshedden committed:
BUG: SAS chunksize / iteration issues (#14743)

closes #14734
closes #13654

jreback committed:
ENH: add data hashing routines (#14729)

xref https://github.com/dask/dask/pull/1807

yarikoptic committed:
BF: (re)raise the exception always unless returning (#14756)

otherwise leads atm to masking of this error while testing on i386
and then failling since

UnboundLocalError: local variable unser referenced before assignment

More detail: https://buildd.debian.org/status/fetch.php?pkg=pandas&arch=i386&ver=0.19.1-1&stamp=1479504883

gfyoung committed:
BUG: Improve error message for skipfooter malformed rows in Python engine (#14749)

Python's native CSV library does not respect the
skipfooter parameter, so if one of those skipped
rows is malformed, it will still raise an error.

Closes gh-13879.

tadiseshan committed:
added read_msgpack() to index (#14765)

jreback committed:
BLD: clean .pxi when cleaning (#14766)

jreback committed:
BLD/DOC: use new secure key for pandas-docs-travis

jreback committed:
ERR: raise on python in object hashing, only supporting strings, nulls

xref #14729

Author: Jeff Reback <jeff@reback.net>

Closes #14767 from jreback/hashing_object and squashes the following commits:

9a5a5d4 [Jeff Reback] ERR: raise on python in object hashing, only supporting strings, nulls

gfyoung committed:
DOC: Remove SparseSeries from SparseArray doc (#14769)

sinhrks committed:
PERF: Improve replace perf

When .replace is called with
`dict`, replacements are done per value. Current impl try to soft
convert the dtype in every replacement, but it is enough to be done in
the final replacement.

Author: sinhrks <sinhrks@gmail.com>

Closes #12745 from sinhrks/replace_perf and squashes the following commits:

ffc59b0 [sinhrks] PERF: Improve replace perf

chris-b1 committed:
DOC/TST: dtype param in read_fwf (#14768)

jorisvandenbossche committed:
PEP8: fix line length

jorisvandenbossche committed:
DOC: specify link to frequencies (#14760)

jreback committed:
BLD: update pandas-docs github token again

jreback committed:
BLD: restore original pandas-docs key

tae-jun committed:
Fix typo at pandas/core/generic.py (#14787)

indentifier -> identifier

dodysw committed:
Fix a simple typo (#14785)

thrasibule committed:
Small typos (#14789)

aileronajay committed:
ENH: support cut/qcut for datetime/timedelta (GH14714) (#14737)

chris-b1 committed:
API: add dtype param to read_excel (#14786)

* API add dtype param to read_excel

* doc fixup

chris-b1 committed:
BUG: multi-index HDFStore data_columns=True

closes #14435

Author: Chris <cbartak@gmail.com>

Closes #14791 from chris-b1/hdf-mi-datacolumns and squashes the following commits:

5d32610 [Chris] BUG: multi-index HDFStore data_columns=True

jreback committed:
BUG: Bug in a groupby of a non-lexsorted MultiIndex

closes #14776

Author: Jeff Reback <jeff@reback.net>

Closes #14777 from jreback/mi_sort and squashes the following commits:

cf31905 [Jeff Reback] BUG: Bug in a groupby of a non-lexsorted MultiIndex and multiple grouping levels

brandonmburroughs committed:
ENH: support kind and na_position kwargs in Series.sort_index (#14445)

swuecho committed:
DOC: minor format fix (#14802)

Commit:
ENH: add Series to_excel method (#14780)

Closes #8825

toobaz committed:
BUG: Ensure min_itemsize is always a list (#11412)

closes #11412

Author: Pietro Battiston <me@pietrobattiston.it>

Closes #14728 from toobaz/minitemsizefix and squashes the following commits:

e25cd1f [Pietro Battiston] Whatsnew
b9bb88f [Pietro Battiston] Tests for previous commit
6406ee8 [Pietro Battiston] BUG: Ensure min_itemsize is always a list

jeffcarey committed:
BUG: Corrects stopping logic when nrows argument is supplied (#7626)

closes #7626

Subsets of tabular files with different "shapes"
will now load when a valid skiprows/nrows is given as an argument   -

Conditions
for error:  1) There are different "shapes" within a tabular data
file, i.e. different numbers of columns.  2) A "narrower" set of
columns is followed by a "wider" (more columns) one, and the narrower
set is laid out such that the end of a 262144-byte block occurs within
it.    Issue summary:   The C engine for parsing files reads in 262144
bytes at a time. Previously, the "start_lines" variable in
tokenizer.c/tokenize_bytes() was set incorrectly to the first line in
that chunk, rather than the overall first row requested. This lead to
incorrect logic on when to stop reading when nrows is supplied by the
user. This always happened but only caused a crash when a wider set of
columns followed in the file. In other cases, extra rows were read in
but then harmlessly discarded.    This pull request always uses the
first requested row for comparisons, so only nrows will be parsed
when supplied.

Author: Jeff Carey <jeff.carey@gmail.com>

Closes #14747 from jeffcarey/fix/7626 and squashes the following commits:

cac1bac [Jeff Carey] Removed duplicative test
6f1965a [Jeff Carey] BUG: Corrects stopping logic when nrows argument is supplied (Fixes #7626)

WillAyd committed:
ENH: Allow usecols to accept callable (GH14154) (#14234)

jreback committed:
COMPAT: numpy compat with 1-ndim object array compat and broadcasting (#14809)

xref #14808

jreback committed:
BUG: fix hash collisions for from int overflow (#14805)

* BUG: we don't like hash collisions in siphash

xref #14767

* This should be a 64-bit int, not an 8-bit int

* fix tests

gfyoung committed:
MAINT: Cleanup pandas/src/parser (#14740)

Remove dead code and reformat for
style using Google's C++ style guide.

Also adds Google's cpplint (fork) to
the style checking for Travis.

mroeschke committed:
TST: Test pivot with categorical data

closes #8860  closes #8731  closes #9534    Looks like the examples in
these issues pass in the master branch. Added test to confirm.

Author: Matt Roeschke <emailformattr@gmail.com>

Closes #14807 from mroeschke/test_pivot_categoricals and squashes the following commits:

b506083 [Matt Roeschke] TST: Test pivot with categorical data

jreback commented:
can you rebase

jreback committed:
BLD: try new build credentials for pandas-docs

mroeschke committed:
DOC: Fix grammar and formatting typos (#14803)

yarikoptic committed:
BF: boost min cython to 0.23

closes #14699
closes #14831
closes #14508

bmagnusson committed:
ENH: Add the ability to have a separate title for each subplot when plotting (#14753)

* Add logic such that if 'title' is a list and 'subplots' is True, use each item of the list as the title of the individual subplots.

Dr-Irv committed:
ENH: Introduce UnsortedIndexError  GH11897 (#14762)

yarikoptic committed:
BF(TST): use = (native) instead of < (little endian) for target data types (#14832)

jreback committed:
BLD: use org name in build-docs.sh

jorisvandenbossche committed:
TST: add test to confirm GH14606 (specify category dtype for empty) (#14752)

Issue #14606 was fixed by PR #14717, adding one more specific test to confirm this

Commit:
DOC: add section on groupby().rolling/expanding/resample (#14801)

aileronajay committed:
ENH: add timedelta as valid type for interpolate with method='time' (#14799)

jorisvandenbossche committed:
CLN/PERF: clean-up of the benchmarks (#14099)

jreback committed:
BLD: try new gh token for pandas-docs

jreback committed:
TST: skip testing on windows for specific formatting which sometimes hangs (#14851)

xref #14626

kordek committed:
BUG: GH11847 Unstack with mixed dtypes coerces everything to object

closes #11847

Changed the way
in which the original data frame is copied (dropped use of .values,
since it does not preserve dtypes).

Author: Pawel Kordek <pawel.kordek@gmail.com>

Closes #14053 from kordek/#11847 and squashes the following commits:

6a381ce [Pawel Kordek] BUG: GH11847 Unstack with mixed dtypes coerces everything to object

Commit:
BUG: Allow TZ-aware DatetimeIndex in merge_asof() (#14844)

closes #14844

Author: Christopher C. Aycock <christopher.aycock@twosigma.com>

Closes #14845 from chrisaycock/GH14844 and squashes the following commits:

97b73a8 [Christopher C. Aycock] BUG: Allow TZ-aware DatetimeIndex in merge_asof() (#14844)

jreback committed:
CLN: lint of test_base.py

mroeschke committed:
Frame benchmarking sum instead of mean (#14824)

mroeschke committed:
TST: Correct results with np.size and crosstab (#4003) (#14755)

missing assert

jreback committed:
BLD: escape GH_TOKEN in build_docs

mroeschke committed:
TST: Test DatetimeIndex weekend offset (#14853)

jreback committed:
BLD: new access token on pandas-dev

jreback committed:
BLD: missing - on secure

xgdgsc committed:
DOC: warning section on memory overflow when joining/merging dataframes on index with duplicate keys (#14788)

closes #14736

toobaz committed:
BUG: Apply min_itemsize to index even when not appending

closes #10381

Author: Pietro Battiston <me@pietrobattiston.it>

Closes #14812 from toobaz/to_hdf_min_itemsize and squashes the following commits:

c07f1e4 [Pietro Battiston] Whatsnew
38b8fcc [Pietro Battiston] Tests for previous commit
c838afa [Pietro Battiston] BUG: set min_itemsize even when there is no need to validate (#10381)

shawnheide committed:
BUG: astype falsely converts inf to integer (GH14265) (#14343)

gr8h committed:
DOC: fix groupby.rst for building issues

closes #14861
closes #14863

patniharshit committed:
DOC: add floats and ints missing as acceptable arguments for pandas.to_datetime (#14864)

dhimmel committed:
CLN: Refactor compression code to expand URL support

closes #14576
closes #12688
closes #14570
xref #14874

jreback committed:
DOC: doc-string for infer_compression

Commit:
ENH: Added errors{'raise','ignore'} for keys not found in meta for json_normalize

Author: dickreuter <dickreuter@yahoo.com>

Closes #14583 from dickreuter/json_normalize_enhancement and squashes the following commits:

701c140 [dickreuter] adjusted formatting
3c94206 [dickreuter] shortened lines to pass linting
2028924 [dickreuter] doc changes
d298588 [dickreuter] Fixed as instructed in pull request page
bcfbf18 [dickreuter] Avoids exception when pandas.io.json.json_normalize

erikcs committed:
ENH/DOC: wide_to_long performance and docstring clarification

closes #14778

Please see
regex search on long columns by first converting to Categorical, avoid
melting all dataframes with all the id variables, and wait with trying
to convert the "time" variable to `int` until last), and clear up the
docstring.

Author: nuffe <erik.cfr@gmail.com>

Closes #14779 from nuffe/wide2longfix and squashes the following commits:

df1edf8 [nuffe] asv_bench: fix indentation and simplify
dc13064 [nuffe] Set docstring to raw literal to allow backslashes to be printed (still had to escape them)
295d1e6 [nuffe] Use pd.Index in doc example
1c49291 [nuffe] Can of course get rid negative lookahead now that suffix is a regex
54c5920 [nuffe] Specify the suffix with a regex
5747a25 [nuffe] ENH/DOC: wide_to_long performance and functionality improvements (#14779)

mroeschke committed:
TST: Parse dates with empty space (#6428) (#14862)

+ Add doc explaining parse_date limitation

adrian-stepien committed:
DOC: Improved links between expanding and cum* (GH12651)

- [x] closes #12651  - [x] passes `git diff upstream/master | flake8
--diff`

Author: adrian-stepien <adrian-stepien@users.noreply.github.com>

Closes #14098 from adrian-stepien/doc/12651 and squashes the following commits:

4427e28 [adrian-stepien] DOC: Improved links between expanding and cum* (#12651)
8466669 [adrian-stepien] DOC: Improved links between expanding and cum* (#12651)
30164f3 [adrian-stepien] DOC: Correct link from b/ffill to fillna

jorisvandenbossche committed:
DOC: update added docs from #14098

jisantuc committed:
BUG: Convert float freqstrs to ints at finer resolution (#14378)

Passing `'0.5min'` as a frequency string should generate 30 second
intervals, rather than five minute intervals. By recursively increasing
resolution until one is found for which the frequency is an integer,
this commit ensures that that's the case for resolutions from days to
microseconds.

Fixes #8419

chrisaycock committed:
ENH: merge_asof() has left_index/right_index and left_by/right_by (#14253) (#14531)

jorisvandenbossche committed:
DOC: already include 0.20 release notes for dev docs

jonmmease committed:
ENH: Allow the groupby by param to handle columns and index levels (GH5677) (#14432)

mroeschke committed:
TST: Same return values in drop_duplicates for Series and DataFrames(#14192) (#14754)

gfyoung committed:
MAINT: Style check *.c and *.h files

`cpplint` was introduced #14740, and this commit extends to check
other `*.c` and `*.h` files.    Currently, they all reside in
`pandas/src`, and this commit expands the lint to check all of the
following:    1) `datetime` (dir)  2) `ujson` (dir)  3)
`period_helper.c`  4) `All header files`    The parser directory was
handled in #14740, and the others have been deliberately omitted per
the discussion <a href="https://github.com/pandas-
dev/pandas/pull/14740#issuecomment-265260209">here</a>.

Author: gfyoung <gfyoung17@gmail.com>

Closes #14814 from gfyoung/c-style-continue and squashes the following commits:

27d4d46 [gfyoung] MAINT: Style check *.c and *.h files

gfyoung committed:
API: Return sparse objects always for cumsum

Always return `SparseArray` and `SparseSeries` for
`SparseArray.cumsum()` and `SparseSeries.cumsum()` respectively,
regardless of `fill_value`.

Closes #12855.

Author: gfyoung <gfyoung17@gmail.com>

Closes #14771 from gfyoung/sparse-return-type and squashes the following commits:

83314fc [gfyoung] API: Return sparse objects always for cumsum

jreback committed:
DOC: doc changes in whatsnew 0.20.0

jreback committed:
DOC: whatsnew 0.19.2

keshavramaswamy committed:
Fixed KDE Plot to drop the missing values (#14820)

BUG: Fixed KDE plot to ignore missing values

 closes #14821

* fixed kde plot to ignore the missing values
* added comment to elaborate the changes made
* added a release note in whatsnew/0.19.2
* added test to check for  missing values and cleaned up whatsnew doc
* added comment to refer the issue
* modified to fit lint checks
* replaced ._xorig with .get_xdata()

jreback committed:
PERF: use StringHasTable for strings

xref #13745

provides a modest speedup for all string hashing. The
key thing is, it will release the GIL on more operations where this is
possible (mainly factorize).
can be easily extended to value_counts() and .duplicated() (for strings)

Author: Jeff Reback <jeff@reback.net>

Closes #14859 from jreback/string and squashes the following commits:

98f46c2 [Jeff Reback] PERF: use StringHashTable for strings in factorizing

ashishsingal1 committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	pandas/tools/tile.py

ashishsingal1 committed:
adding test and moving unique bins code.

jorisvandenbossche commented:
@ashishsingal1 something went wrong with your rebase. Can you do:

```
git fetch upstream
git rebase upstream/master
git push -f
```

That should normally solve it

ashishsingal1 commented:
Trouble rebasing, going to start over with a new PR.
